Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm

```diff
@@ -750,17 +751,33 @@ static NavigatingToAppBoundDomain schemeOrDomainIsAppBound(const String& host, c
             return;
         static const auto maxManagedDomainCount = 10;
         NSArray<NSString *> *crossSiteTrackingPreventionRelaxedDomains = nil;
+        NSArray<NSString *> *crossSiteTrackingPreventionRelaxedApps = nil;
+
+        bool isSafari = false;
 #if PLATFORM(MAC)
+        isSafari = WebCore::MacApplication::isSafari();
         NSDictionary *managedSitesPrefs = [NSDictionary dictionaryWithContentsOfFile:[[NSString stringWithFormat:@"/Library/Managed Preferences/%@/%@.plist", NSUserName(), kManagedSitesIdentifier] stringByStandardizingPath]];
         crossSiteTrackingPreventionRelaxedDomains = [managedSitesPrefs objectForKey:kCrossSiteTrackingPreventionRelaxedDomainsKey];
+        crossSiteTrackingPreventionRelaxedApps = [managedSitesPrefs objectForKey:kCrossSiteTrackingPreventionRelaxedAppsKey];
 #elif !PLATFORM(MACCATALYST)
+        isSafari = WebCore::IOSApplication::isMobileSafari();
         if ([PAL::getMCProfileConnectionClass() instancesRespondToSelector:@selector(crossSiteTrackingPreventionRelaxedDomains)])
             crossSiteTrackingPreventionRelaxedDomains = [(MCProfileConnection *)[PAL::getMCProfileConnectionClass() sharedConnection] crossSiteTrackingPreventionRelaxedDomains];
         else
             crossSiteTrackingPreventionRelaxedDomains = @[];
+
+        auto relaxedAppsSelector = NSSelectorFromString(@"crossSiteTrackingPreventionRelaxedApps");
+        if ([PAL::getMCProfileConnectionClass() instancesRespondToSelector:relaxedAppsSelector])
+            crossSiteTrackingPreventionRelaxedApps = [[PAL::getMCProfileConnectionClass() sharedConnection] performSelector:relaxedAppsSelector];
+        else
+            crossSiteTrackingPreventionRelaxedApps = @[];
 #endif
         managedKeyExists = !!crossSiteTrackingPreventionRelaxedDomains;
     
+        NSString *bundleID = [[NSBundle mainBundle] bundleIdentifier];
+        if (!isSafari && !isRunningTest(bundleID) && ![crossSiteTrackingPreventionRelaxedApps containsObject:bundleID])
```

Comment:
I always find these compound clauses hard to read and check for correctness, especially when they contain the not operator. My suggestion is to add a convenience variable `bool isAllowedApplication = isSafari || [crossSiteTrackingPreventionRelaxedApps containsObject:bundleID] || isRunningTest(bundleID)` and then write the test as `if (!isAllowedApplication)`.